lib/spack/spack/package.py

```diff
@@ -983,28 +977,42 @@ def module(self):
                           fromlist=[self.__class__.__name__])
 
 
-    def setup_dependent_environment(self, module, spec, dependent_spec):
-        """Called before the install() method of dependents.
+    def environment_modifications(self, dependent_spec):
```

Comment:
@tgamblin Very reasonable. We may in fact have a default implementation for `setup_dependent_environment` that delegates to `setup_environment`, so that in all the cases where the modifications are exactly the same, the user can implement just one of the two functions.

Regarding the parameter vs. return value : I usually prefer the second way of doing things, but I see the point here. I'll make these changes asap.